sdk/spring/spring-cloud-azure-autoconfigure/src/main/java/com/azure/spring/cloud/autoconfigure/aad/configuration/AadOAuth2ClientConfiguration.java

```diff
@@ -3,174 +3,160 @@
-            manager.setAuthorizedClientProvider(authorizedClientProviders);
-            return manager;
-        }
+    @Bean
+    @ConditionalOnMissingBean
+    public OAuth2ClientAuthenticationJWKResolver oAuth2ClientAuthenticationJWKResolver(AadAuthenticationProperties properties) {
+        return new AadOAuth2ClientAuthenticationJWKResolver(properties.getCredential().getClientCertificatePath(),
+            properties.getCredential().getClientCertificatePassword());
     }
 
     /**
-     * Web application and resource server scenario, OAuth2AuthorizedClientManager configuration for AAD.
+     * Declare OAuth2AuthorizedClientManager bean.
+     *
+     * @param clientRegistrations the client registration repository
+     * @param authorizedClients the OAuth2 authorized client repository
+     * @param jwkResolver the {@link JWK} function resolver
+     * @return OAuth2AuthorizedClientManager bean
      */
-    @Configuration(proxyBeanMethods = false)
-    @Conditional(WebApplicationAndResourceServerCondition.class)
-    public static class WebApplicationAndResourceServiceOAuth2AuthorizedClientManagerConfiguration {
+    @Bean
+    @ConditionalOnMissingBean
+    public OAuth2AuthorizedClientManager authorizedClientManager(ClientRegistrationRepository clientRegistrations,
+                                                                 OAuth2AuthorizedClientRepository authorizedClients,
+                                                                 OAuth2ClientAuthenticationJWKResolver jwkResolver) {
+        DefaultOAuth2AuthorizedClientManager manager =
+            new DefaultOAuth2AuthorizedClientManager(clientRegistrations, authorizedClients);
+        Function<ClientRegistration, JWK> jwkFunction = jwkResolver.resolve();
+        RefreshTokenOAuth2AuthorizedClientProvider refreshTokenProvider = getRefreshTokenProvider(jwkFunction);
+        AadAzureDelegatedOAuth2AuthorizedClientProvider azureDelegatedProvider =
+            new AadAzureDelegatedOAuth2AuthorizedClientProvider(refreshTokenProvider, authorizedClients);
+        JwtBearerOAuth2AuthorizedClientProvider jwtBearerProvider =
+            getJwtBearerProvider(jwkFunction);
+        // @formatter:off
+        OAuth2AuthorizedClientProvider providers =
+            OAuth2AuthorizedClientProviderBuilder
+                .builder()
+                .authorizationCode()
+                .clientCredentials(builder -> clientCredentialsBuilderConsumer(jwkFunction, builder))
+                .password(builder -> passwordBuilderConsumer(jwkFunction, builder))
+                .provider(refreshTokenProvider)
+                .provider(jwtBearerProvider)
+                .provider(azureDelegatedProvider)
+                .build();
+        // @formatter:on
+        manager.setAuthorizedClientProvider(providers);
+        return manager;
+    }
+
+    private void passwordBuilderConsumer(Function<ClientRegistration, JWK> jwkFunction,
```

Comment:
How about this method just return a `accessTokenResponseClient`, and call the customization in line 111?